### PR TITLE
Add backbone of refund

### PR DESCRIPTION
**Pull request by muratovv**
Add pipeline of refund.
There are 3 steps:
* query to Iroha
* refund creation
* and signing
Each step is potentially separated task. Also, in steps implementation, we will separate those steps by own class with dependencies and interfaces.

### CODE REVIEW

File: src/main/kotlin/endpoint/eth/EthNotaryResponse.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ sealed class EthNotaryResponse {
      */
     data class Successful(
         val ethSignature: EthSignature,
-        val ethPublicKey: EthPublicKey
+        val ethRefund: EthRefund
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/endpoint/eth/EthNotaryResponse.kt]: please, add documentation for data fields

File: src/main/kotlin/endpoint/eth/EthRefund.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Type of coin in Ethereum. Possibly it will be an address
+ */
+typealias CoinType = String
+
+/**
+ * Type of amount in Ethereum
+ */
+typealias AmountType = Double
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/endpoint/eth/EthRefund.kt]: I think amount in ether is BigInteger. It seems to me that it is better to move all Ethereum data types into separate file.

File: src/main/kotlin/endpoint/eth/EthRefundStrategyImpl.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * The method passes the query with transaction to Iroha and return the answer
+     * @param request - user's request with transaction hash
+     * @return Transaction which is appeared in Iroha or error
+     */
+    private fun performQuery(request: EthRefundRequest): Result<iroha.protocol.BlockOuterClass.Transaction, NotaryException> {
+        val query = ModelQueryBuilder()
+        val hashes = HashVector()
+        HashVector().add(Hash(request.irohaTx))
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/endpoint/eth/EthRefundStrategyImpl.kt]: mb hashes.add()?

Comment (muratovv) [reply src/main/kotlin/endpoint/eth/EthRefundStrategyImpl.kt]: What do you mean?

File: src/main/kotlin/endpoint/eth/EthRefund.kt (hunk 2 above)
Comment (muratovv) [reply src/main/kotlin/endpoint/eth/EthRefund.kt]: Yeah, we have to investigate the type on eth side and put it here.

File: src/main/kotlin/notary/NotaryInitialization.kt [hunk 4]
```diff
@@ -142,16 +143,13 @@ class NotaryInitialization {
...
             ServerInitializationBundle(CONFIG[ConfigKeys.refundPort], CONFIG[ConfigKeys.ethEndpoint]),
             mock {
-                val request = any<EthRefundContract>()
+                val request = any<EthRefundRequest>()
                 on {
                     performRefund(request)
                 } doReturn EthNotaryResponse.Successful(
                     "signature",
-                    "pub_key"
+                    EthRefund("address", "coin", 66.6)
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/notary/NotaryInitialization.kt]: Currently amount is BigInteger, so `BigInteger.valueOf(666)`

Comment (muratovv) [reply src/main/kotlin/notary/NotaryInitialization.kt]: The issue has been fixed already.

**Pull request merged**